### ISSUE TITLE
Propagate labels from ServiceBinding to the generated Secret

### ISSUE THREAD

**Issue by pre** [SAP/sap-btp-service-operator#360]
We have an IAS `ServiceBinding` providing credentials to ArgoCD via a `Secret` resource. 

The ArgoCD configuration requires the `Secret` be labeled with 
```
  labels:
    app.kubernetes.io/part-of: argocd
```

> NOTE: Secret must have label `app.kubernetes.io/part-of: argocd`
https://argo-cd.readthedocs.io/en/stable/operator-manual/user-management/#alternative

It would be really simple, if we could add these labels to the `ServiceBinding` from where sap-btp-service-operator would copy these labels to the generated `Secret`.  

While we could add these labels "by hand" to the generated `Secret`, we wish not to do so. It feels unsafe / unmaintainable since there is no guarantee that these labels stay there when sap-btp-service-operator rotates the credentials, or if behaviour of sap-btp-service-operator changes in a future release. 

By and large, sap-btp-service-operator as a controller owns the `Secret` which it generates, and sap-btp-service-operator should be the sole actor modifying its contents. Anyway, should we add these labels externally from sap-btp-operator to the generated `Secret`, it would require us to define the `Secret` (generated by sap-btp-service-operator) as a tracked resource in ArgoCD which in turn means ArgoCD should be set to ignore everything else than `metadata.labels` of that Secret - which in turn sounds like a configuration bloat. 

We have an overly complicated workaround to provide the labels:
* sap-btp-service-operator creates the `Secret` "original-secret"  from the `ServiceBinding` 
* external-secrets-operator copies contents of "original-secret" and creates a new secret "secret-with-labels".
* ArgoCD uses "secret-with-labels" 
* When sap-btp-service-operator rotates credentials, it modifies "original-secret"
* When "original-secret" is modified, external-secrets-operator copies the changes to "secret-with-labels". 

In short, it would be neat and less complex if we would get the labels directly to "original-secret" by sap-btp-service-operator.

Previous issues:
* https://github.com/SAP/sap-btp-service-operator/issues/333
* https://github.com/SAP/sap-btp-service-operator/issues/259
* https://github.com/SAP/sap-btp-service-operator/issues/307

**Comment by pre** [comment]
Please don’t mark the issue as stale. Could you reopen @kerenlahav

**Comment by kerenlahav** [comment]
released in v0.6.1

**Comment by loewenstein** [comment]
Question @kerenlahav: looking at the implementation and the requested feature of adding to metadata, I wonder if it would be possible to leave out `stringData` to trigger the controller to deliver the original generated Secret content. What do you think?
